### ISSUE TITLE
[iOS][New Arch][Expo Router] Bottom of screen in presentation: "modal" is cropped

### ISSUE THREAD

**Issue by delphinebugner** [expo/expo#32450]
### Minimal reproducible example

https://github.com/delphinebugner/expo-test-52

### What platform(s) does this occur on?

iOS

### Where did you reproduce the issue?

in a development build, in Expo Go, in a standalone app

## Description
TL;DR
![image](https://github.com/user-attachments/assets/d76bea3f-afaf-4b3c-8dcf-ab46d055052c)

When using `screenPresentation: "modal"` on iOS, since New Arch, bottom of the screen is cropped. 

You can try it on the repro app by clicking the "Open Menu" button. There's a text at the bottom of the screen that is not visible.

Note that it does *not* occur without expo and expo router, with just React Navigation + React Native Screens set-up. This is why I'm opening the issue here 🙂 

## Where does it occur

According what I tried: 

| Architecture | Navigation | RN / Expo Version | Result |
|--------|--------|--------|--------|
| Old Arch | Expo Router | RN 0.75 + Expo 51 | No error ✅  |
| New Arch | Expo Router | RN 0.75 + Expo 51 (my app) | Error 😵  |
| New Arch | Expo Router | RN 0.76 + Expo 52 beta (the repro example) | Error 😵 |
| New Arch | React Navigation | RN 0.75 / RN 0.76 | No error ✅ (link for a repro with RN 0.76 : https://github.com/delphinebugner/test76)  |

### Environment

```text
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.6.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 21.6.1 - ~/.asdf/installs/nodejs/21.6.1/bin/node
      Yarn: 1.22.22 - ~/.asdf/installs/nodejs/21.6.1/bin/yarn
      npm: 10.2.4 - ~/.asdf/plugins/nodejs/shims/npm
      Watchman: 2024.03.25.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /Users/delphinebugner/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.0, iOS 18.0, macOS 15.0, tvOS 18.0, visionOS 2.0, watchOS 11.0
    IDEs:
      Android Studio: 2024.1 AI-241.18034.62.2412.12266719
      Xcode: 16.0/16A242d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.0-preview.10 => 52.0.0-preview.10
      expo-router: ~4.0.0-preview.6 => 4.0.0-preview.6
      react: 18.3.1 => 18.3.1
      react-dom: 18.3.1 => 18.3.1
      react-native: 0.76.0 => 0.76.0
      react-native-web: ~0.19.13 => 0.19.13
    npmGlobalPackages:
      eas-cli: 12.6.0
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check if the project meets version requirements for submission to app stores
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check npm/ yarn versions
✔ Check for issues with Metro config
✔ Validate packages against React Native Directory package metadata
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK
```

**Comment by chrfalch** [comment]
Hi @delphinebugner - thanks for submitting this issue. 

I did some testing and it might seem like this is not related to Expo Router but to RNScreens 4.0.0 (beta). RNScreens already has an issue describing the problem: 

https://github.com/software-mansion/react-native-screens/issues/2455

I tested removing Expo-Router completely from the demo-project you [provided](https://github.com/delphinebugner/expo-test-52) and the issue is still there. I also reduce the example to just be a Stack and that modal screen - still seeing the same issue. Finally I created a new bare react-native project using RN 0.76 and added React Navigation version 7 and got the same problem.

I think the difference between the two project you included is that the [second example](https://github.com/delphinebugner/test76) (without Expo Router) uses an earlier version of React-navigation (version 6) which doesn't use RNScreens 4.0.

**Comment by delphinebugner** [comment]
Hi @chrfalch , thanks for looking into it 🙂 
What is weird is on my real app, I use RN Screen `3.34.0` and I have the issue, so it's not just the combination RN Screen 4 + New Arch ... Even if I agree it looks more like an RN Screen issue than a Router oone :) let's update the recap table with your cases and mine :

### Recap

| Architecture | Navigation | RN / Expo Version | RN Screen Version | Result | Tester |
|--------|--------|--------|--------|-------|-------|
| Old Arch | Expo Router | RN 0.75 + Expo 51 | 3.34.0 | No error ✅  | Delphine |
| New Arch | Expo Router | RN 0.75 + Expo 51 (my app) | 3.34.0 / 3.35.0 | Error 😵  | Delphine |
| New Arch | Expo Router | RN 0.76 + Expo 52 beta (the repro example) | 4.0.0-beta.14 | Error 😵 | Delphine |
| New Arch | Expo Without Expo Router | RN 0.76 + Expo 52 beta (Christian's test) | 4.0.0-beta.14  |  Error 😵 | Christian |
| New Arch | React Navigation 6 | RN 0.75 / RN 0.76 | 3.35.0 | No error ✅ [repro RN 0.76]( https://github.com/delphinebugner/test76)  | Delphine |
| New Arch | React Navigation 7 | RN 0.76 | 4 beta? | Error 😵  | Christian |

**Comment by chrfalch** [comment]
I've made a reproduction using bare RN 0.76 that shows the error. The strange thing is that I'm able to narrow it down to using an Animated.View (REA) with an animated style. Removing the animated style removes the error above...

This repro contains all the details:

https://github.com/chrfalch/screens-repro

**Comment by chrfalch** [comment]
This issue is related to Reanimated and not to RN Screens. I found a similar [issue](https://github.com/software-mansion/react-native-reanimated/issues/6659) in REA now and did some investigation:

I reported this in the RNScreens [issue](https://github.com/software-mansion/react-native-screens/issues/2455#issuecomment-2450385477) that this was working when using `REA 3.14.0` and failed starting from `REA 3.15.0`.

After investigating the commits [between the two releases](https://github.com/software-mansion/react-native-reanimated/compare/3.14.0...3.15.0) I was able to find the offending PR here:

https://github.com/software-mansion/react-native-reanimated/pull/6214

This one is containing an optimization to how shadow nodes are updated and if we revert this optimization (even tried reverting it from 3.16.1 by doing some manual copy/paste) the issues is no longer happening.

**Comment by delphinebugner** [comment]
Impressive investigation 😮 that explains the differences, I did'nt use Reanimated on my 0.76 test, when you had it with the expo version without expo router! 

Do we close this issue, and open one on Reanimated?

**Comment by chrfalch** [comment]
Yeah, it was a bit deep down - and the reason even though you weren't using REA was that Expo Router which uses React Navigation which uses Reanimated 😅

I've already moved the case to the reanimated repo using an already existing [issue](https://github.com/software-mansion/react-native-reanimated/issues/6659). Will keep this one open to track progress here / SDK-52.

**Comment by Demven** [comment]
I have the same issue (only on iOS), and can't find ways to bypass it. Hoping it can be fixed soon 🙏🏻 .

**Comment by maksim-romanov** [comment]
same issue for me, IOS only
